test/cql-pytest/test_query_validation.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+from cassandra.protocol import InvalidRequest
+from util import unique_name, unique_key_string
+
+
+@pytest.fixture(scope="module")
+def table1(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (id text, a int, b int, c list<int>, PRIMARY KEY((id),a,b))")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+
+@pytest.fixture(scope="module")
+def table2(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (id1 text, id2 text, a int, c list<int>, PRIMARY KEY((id1,id2),a))")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+
+def test_insert_with_list_column_and_missing_clustering_key_part(cql, table1):
+    key = unique_key_string()
+    with pytest.raises(InvalidRequest, match='Missing mandatory PRIMARY KEY part a'):
+        cql.execute(cql.prepare(f"INSERT INTO {table1} (id,b,c) VALUES ('{key}',1,null)"))
```

Comment:
When I read the commit message, I thought to myself that I didn't understand why it is specific to collection columns. But now, reading this test, I had a thought:
With non-frozen collections, an **empty list** is identical to a null (a missing list). So maybe a null list was intentionally allowed - to support the ability to have an empty list? Or, maybe the other way around - and empty list should not be allowed? I have to admit I'm not sure. I'm not sure I even knew that it was legal to have a non-frozen collection as part of a key! In any case, the first thing I would check is what Cassandra does in this case - and this is why we have "run-cassandra" that I asked you above to run.
In any case, if the purpose of this test is to check this meaning of a null collection as part of a key, it would be good to explain this in a comment above the test.

Reply:
Also, maybe in the same test you should also try a null for "b", which should also fail with an error? Even if we didn't have a bug in this case (and only had a bug in the case of "c"), we should have a regression test to check such a bug doesn't suddenly happen in the future.